### PR TITLE
Client cleanup - include all client columns

### PR DESCRIPTION
**Pull request by eanders**
## _Merging this PR_
- use the squash-merge strategy for PRs targeting a release-X branch
- use a merge-commit or rebase strategy for PRs targeting the stable branch

## Description
[//]: # (Summarize changes and include links related issue. List any new dependencies)
This updates `ClientCleanup` to include all (or almost all) client columns when setting values on the destination client.  Specifically, we were never setting some veteran related values which are needed for SSVF reporting and weren't being exported in the HMIS CSV exporter.  Additionally, `AdditionalRaceEthnicity` and `DifferentIdentityText` are included to make the exporter more complete.

## Type of change
[//]: # (e.g., Bug fix, New feature, Code clean-up, Dependency update)
Bug fix

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [ ] If adding a new endpoint / exposing data in a new way, I have:
  - [ ] ensured the API can't leak data from other data sources
  - [ ] ensured this does not introduce N+1s
  - [ ] ensured permissions and visibility checks are performed in the right places
- [x] I have updated the documentation (or not applicable)
- [x] I have added spec tests (or not applicable)
- [x] I have provided testing instructions in this PR or the related issue (or not applicable)

### CODE REVIEW

File: app/models/grda_warehouse/tasks/client_cleanup.rb [hunk 1]
```diff
@@ -471,6 +550,14 @@ def choose_best_gender dest_attr, source_clients
         end
       end
 
+      # Set DifferentIdentityText
+      additional = sorted_source_clients.map { |m| m[:DifferentIdentityText] }.compact
+      dest_attr[:DifferentIdentityText] = if additional.empty?
+        nil
+      else
+        additional.first
+      end
```
Comment (ttoomey) [inline app/models/grda_warehouse/tasks/client_cleanup.rb]: do we need to handle blank strings (s/compact/compact_blank/)? Also maybe could golf down the assignment conditional
```suggestion
      additional = sorted_source_clients.map { |m| m[:DifferentIdentityText] }.compact_blank
      dest_attr[:DifferentIdentityText] = additional.presence&.first
```

File: app/models/grda_warehouse/tasks/client_cleanup.rb [hunk 2]
```diff
@@ -414,12 +459,45 @@ def choose_best_ssn dest_attr, source_clients
       dest_attr
     end
 
+    # Set the veteran status based on the calculation in VeteranStatusCalculator
+    # If the client is a veteran, pull veteran related columns from the most-recent veteran source client
+    # If the client isn't a veteran, blank the associated veteran columns
     def choose_best_veteran_status dest_attr, source_clients
-      dest_attr[:VeteranStatus] = calculate_best_veteran_status(dest_attr[:verified_veteran_status], dest_attr[:va_verified_veteran], source_clients)
+      # Sort in reverse chronological order (newest first)
+      sorted_source_clients = source_clients.sort { |a, b| b[:DateUpdated] <=> a[:DateUpdated] }
```
Comment (ttoomey) [inline app/models/grda_warehouse/tasks/client_cleanup.rb:467]: can DateUpdated ever be null? If so this comparison might crash

**Review by ttoomey — APPROVED** [review]
looks good!

File: app/models/grda_warehouse/tasks/client_cleanup.rb (hunk 2 above)
Comment (eanders) [reply app/models/grda_warehouse/tasks/client_cleanup.rb:467]: Hmm, technically yes, though the spec doesn't allow it.  We've been doing this same sort on line 528 for at least 2 years, so I think it's ok.

**Pull request merged**